Jonny Hicks

KidFromCalifornia opened the pull request:
site: 
https://wehappy.netlify.app/

KidFromCalifornia committed:
Implement core functionality for Happy Thoughts app with message submission, display, and styling

KidFromCalifornia committed:
Add functionality for fetching and liking thoughts, enhance UI components, and implement footer

KidFromCalifornia committed:
Add onLike prop to MessageList component for handling likes- allowing buttons to work again

KidFromCalifornia committed:
Add loading state and spinner for fetching thoughts in App component

KidFromCalifornia committed:
Enhance App component with error handling, add header for accessibility, and improve MessageList rendering. Update styles for message items with animations and responsive design.

KidFromCalifornia committed:
Update .gitignore, fix favicon link in index.html, enhance error handling in handleLike, simplify MessageItem component, and improve CSS transitions for message items.

KidFromCalifornia committed:
Refactor MessageItem and MessageList components for improved structure and styling. Update header text for consistency, adjust CSS transitions for message items, and ensure animations work

solen80a commented on src/Components/HandlesLikes.jsx:
I noticed that you used .then method to fetch the API in App.jsx and the async method here. Try to selecet one of the methods for consistency.

solen80a commented on src/App.jsx:
```diff
@@ -1,5 +1,76 @@
+      </header>
+      <main>
+        {loading ? (
+          <LoadingSpinner /> // Show loading message while fetching
```
It look really nice with the loader positioned here when ,oading the page.

simfrisk replied:
I agree!

solen80a reviewed (commented):
Hi Jonny, really nice work with you Happy thoughts site.
Like that you boke out all the styling to own components, easy to read and understand.

I hade some issues to like my posted comment, even after refresh. But after I added a like in postman it worked to like. So seems to be some issue with the POST of likes.

You should be proud and so cute pounding loader, made me so happy 😍

KidFromCalifornia committed:
Remove disabled attribute from HeartButton in MessageItem component to allow liking messages regardless of heart count.

KidFromCalifornia committed:
Update API endpoint

HIPPIEKICK reviewed (changes requested):
Please update your app with this new link (you can use cmd+shift+F to search for all the instances in a repo - you didn’t update on all places).

https://happy-thoughts-api-4ful.onrender.com/

In the meantime, a note about the structure: folders that group things together like components should be named in lowercase. If you have a folder for each component on the other hand, it should be named with uppercase since it should be named the same as the actual component, e.g. Footer.

And I’d probably split the components and the styled components into two different folders, and call the one for styled components ui, styles or similar.

KidFromCalifornia committed:
Update pull request template with Netlify link

KidFromCalifornia committed:
Refactor component imports and styles; update CSS file structure for better organization.

KidFromCalifornia committed:
Refactor styles: move components from Boxstyles to Messagestyles and update imports in MessageForm and MessageItem

HIPPIEKICK commented:
I don't think you have continuous deployment on Netlify, because it's still trying to fetch from the old link 👀

KidFromCalifornia committed:
resetting file naming error

KidFromCalifornia committed:
reset  cache attempting to fix  naming and case sensitivity issues

KidFromCalifornia committed:
Fix: ensure word-break is applied consistently in TextAreaStyle

KidFromCalifornia committed:
Temporarily rename components folder to fix casing

KidFromCalifornia committed:
Fix casing for components folder

KidFromCalifornia committed:
trying again

KidFromCalifornia committed:
Fix import path for handleLike component

KidFromCalifornia commented:
@HIPPIEKICK There was an issue with case sensitivity in the folder names, which caused the builds to keep failing since 10:30 this morning. I had to go to each path, change the folder names, and reset them. After that, I deleted and redeployed the builds. I believe it is working now.

simfrisk commented on src/App.jsx:
```diff
@@ -1,5 +1,76 @@
+      })
+      .catch((error) => {
+        console.error(
+          "Oh no! Previous thoughts are unavailable currently:",
```
Nce error handeling

simfrisk commented on src/components/LoadingSpinner.jsx:
Like Sofia said! Great loader! Alway love your design and idéa!

simfrisk commented on src/components/MessageForm.jsx:
```diff
@@ -0,0 +1,62 @@
+    setMessage(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
```
Good and clear naming

simfrisk commented on src/styles/index.css:
Looks really great

simfrisk reviewed (commented):
Love your site Jonny! You followed the deign great, everything works as expected and I really think you did a great job. Really love the loader even though for some reason it loads for a long time for me. I think you did the assigment great with clear and clean code! Well done!

HIPPIEKICK commented on src/components/MessageForm.jsx:
```diff
@@ -0,0 +1,62 @@
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setLoading(true); // Set loading state to true
+    fetch("https://happy-thoughts-api-4ful.onrender.com//thoughts", {
```
You got an extra slash here that shouldn't be here, it's breaking things so right now it's not possible to POST.

KidFromCalifornia committed:
attempted to optimsing page loading  speeds

KidFromCalifornia committed:
Fix duplicate slash in fetch URL for posting thoughts

HIPPIEKICK reviewed (approved):
⭐

KidFromCalifornia committed:
Fix fetch URL to point to local server for thoughts

KidFromCalifornia committed:
Update API URLs to use production endpoint and fix preload link in index.html now linked to render

KidFromCalifornia committed:
Fix duplicate slash in fetch URL for liking thoughts

KidFromCalifornia committed:
Refactor code structure for improved readability and maintainability

KidFromCalifornia committed:
Fix incorrect paths for CSS files in index.html

KidFromCalifornia committed:
Add user authentication and search functionality with styled components

- Implemented LoginForm component for user login and signup
- Added NavBar with search functionality and login/logout options
- Integrated search feature in SearchBar component
- Updated API calls to include authorization tokens
- Enhanced UI with styled components for better user experience

KidFromCalifornia committed:
Enhance login and message submission functionality with loading states and access token validation

KidFromCalifornia committed:
Add flex-wrap to NavStyle for improved responsiveness

KidFromCalifornia committed:
Refactor SearchBar component: rename file to match component name and move implementation to SearchBarCom.jsx

KidFromCalifornia committed:
Rename Searchbar.jsx to SearchBar.jsx (fix case)

KidFromCalifornia committed:
Remove SearchBar component and its associated styles

KidFromCalifornia committed:
Enhance MessageForm with access token validation and add edit functionality to MessageItem

assigned username to post

KidFromCalifornia committed:
Implement edit and delete functionality for thoughts, including username handling in MessageForm and MessageItem components.
upadates UI with out  page

KidFromCalifornia committed:
Add SweetAlert2 integration and enhance user feedback in forms and message interactions

KidFromCalifornia committed:
Refactor user authentication handling: unify localStorage key casing, pass setCurrentUser to components, and ensure isLoggedIn state is correctly managed.

KidFromCalifornia committed:
Refactor code structure for improved readability and maintainability

KidFromCalifornia committed:
Implement code changes to enhance functionality and improve performance
